xcdat/spatial.py

```diff
@@ -729,11 +743,45 @@ def _averager(self, data_var: xr.DataArray, axis: List[SpatialAxis]):
         """
         weights = self._weights.fillna(0)
 
+        # ensure required weight is between 0 and 1
+        if minimum_weight is None:
+            minimum_weight = 0.0
+        elif minimum_weight < 0.0:
+            raise ValueError(
+                "minimum_weight argument is less than 0. "
+                "minimum_weight must be between 0 and 1."
+            )
+        elif minimum_weight > 1.0:
+            raise ValueError(
+                "minimum_weight argument is greater than 1. "
+                "minimum_weight must be between 0 and 1."
+            )
+
+        # need weights to match data_var dimensionality
+        if minimum_weight > 0.0:
+            weights, data_var = xr.broadcast(weights, data_var)
```

Comment:
I think we can remove this line.

Per docs for [xr.broadcast](https://docs.xarray.dev/en/stable/generated/xarray.broadcast.html):

xarray objects automatically broadcast against each other in arithmetic operations, so this function should not be necessary for normal use.

Reply:
I **think** the weights are probably of dimensions `[lat, lon]` and this casts them to `[time, lat, lon]`. That way when you sum them below you get `[time]` (instead of a dimensionless scalar) which is then used to figure out which values do not meet the weight threshold and need to be masked.

Reply:
Xarray should be handling the automatic broadcasting when using the `xr.where()` API to mask the data using weights. Otherwise, it would probably break with an error that the dimensions don't align. 

According to the [docs](https://docs.xarray.dev/en/stable/generated/xarray.where.html) for `xr.where()`:

> Performs xarray-like broadcasting across input arguments.

Reply:
The unit tests pass with this line commented out. I also didn't implement `xr.broadcast()` explicitly in PR #683.